client/my-sites/plugins/plans/index.tsx

```diff
@@ -40,6 +42,18 @@ const Plans = () => {
 		);
 	}, [ dispatch, translate, selectedSite, breadcrumbs.length ] );
 
+	const getIntervalType = () => {
+		const urlParts = getUrlParts( typeof window !== 'undefined' ? window.location?.href : '' );
+		const intervalType = urlParts?.searchParams.get( 'intervalType' ) || '';
+
+		if ( [ 'yearly', 'monthly' ].includes( intervalType ) ) {
+			return intervalType;
+		}
+
+		// Default value
+		return 'yearly';
+	};
```

Comment:
We can get this out of route params. e.g. modify [our new route](https://github.com/Automattic/wp-calypso/blob/5ee25a2101edd07682c9be86ecec84acb322385e/client/my-sites/plugins/index.web.js#L93) to something like: `/plans/:site/:interval?` (default to yearly). We'll need to update the links in the toggle for this.

Reply:
I like that the `timeInterval` is part of route params since it seems more like an option to be toggled instead of a new route.

I might be missing context on why we would want to get it out of route params.

Reply:
Was purely going by the plan page behavior but to answer why would we want to get a route param out of router params?

I think the main reason is because the router is geared up to manage url inputs like this already, we don't need the extra code here to manage what is an interval query param. We also don't need to know where our inputs are coming from here (window.location), we just need the prop and can defer figuring all that out to the caller.

In the general case our components shouldn't care about window context unless it's really not possible to avoid it.